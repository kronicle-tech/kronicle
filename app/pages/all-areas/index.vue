<template>
  <div>
    <b-container fluid>
      <b-row>
        <b-col>
          <AllAreasTabs />

          <b-alert show="10" dismissible variant="info" class="my-3">
            Click an area's name in the table below to view more information about
            that area
          </b-alert>

          <table
            class="table table-dark table-bordered table-striped mt-2"
            style="width: 100%"
          >
            <thead>
              <tr>
                <th class="area-name">Name</th>
                <th>Description</th>
              </tr>
            </thead>
            <tbody>
              <tr v-for="area in areas" :key="area.id">
                <td class="area-name table-primary">
                  <AreaName :area="area" />
                </td>
                <td>
                  <Markdown :markdown="area.description" />
                </td>
              </tr>
            </tbody>
          </table>
        </b-col>
      </b-row>
    </b-container>
  </div>
</template>

<script lang="ts">
import Vue from 'vue'
import { MetaInfo } from 'vue-meta'
import {BAlert, BCol, BContainer, BRow} from 'bootstrap-vue'
import { Area } from '~/types/kronicle-service'
import AllAreasTabs from '~/components/AllAreasTabs.vue'
import AreaName from '~/components/AreaName.vue'
import Markdown from '~/components/Markdown.vue'

export default Vue.extend({
  components: {
    AllAreasTabs,
    'b-alert': BAlert,
    'b-col': BCol,
    'b-container': BContainer,
    'b-row': BRow,
    AreaName,
    Markdown,
  },
  async asyncData({ $config }) {
    const areas = await fetch(
      `${$config.serviceBaseUrl}/v1/areas?fields=areas(id,name,description)`
    )
      .then((res) => res.json())
      .then((json) => json.areas as Area[])

    return {
      areas,
    }
  },
  data() {
    return {
      areas: [] as Area[],
    }
  },
  head(): MetaInfo {
    return {
      title: 'Kronicle - All Areas',
    }
  },
})
</script>
