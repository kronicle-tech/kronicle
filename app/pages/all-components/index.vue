<template>
  <div>
    <AllComponentsTabs />
    <b-alert show="10" dismissible variant="info" class="my-3">
      Click a component's name in the table below to view more information about
      that component
    </b-alert>

    <ComponentsView :components="components" />
  </div>
</template>

<script lang="ts">
import Vue from 'vue'
import { BAlert } from 'bootstrap-vue'
import { MetaInfo } from 'vue-meta'
import { Component } from '~/types/kronicle-service'
import AllComponentsTabs from '~/components/AllComponentsTabs.vue'
import ComponentsView from '~/components/ComponentsView.vue'

export default Vue.extend({
  components: {
    AllComponentsTabs,
    'b-alert': BAlert,
    ComponentsView,
  },
  async asyncData({ $config, route, store }) {
    const components = await fetch(
      `${$config.serviceBaseUrl}/v1/components?fields=components(id,name,typeId,tags,description,notes,responsibilities,teams,platformId)`
    )
      .then((res) => res.json())
      .then((json) => json.components as Component[])

    store.commit('componentFilters/initialize', {
      components,
      route,
    })

    return {
      components,
    }
  },
  data() {
    return {
      components: [] as Component[],
    }
  },
  head(): MetaInfo {
    return {
      title: 'Kronicle - All Components',
    }
  },
})
</script>
