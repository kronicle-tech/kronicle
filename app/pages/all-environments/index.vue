<template>
  <div class="m-3">
    <EnvironmentStatesView :components="components" />
  </div>
</template>

<script lang="ts">
import Vue from 'vue'
import { MetaInfo } from 'vue-meta'
import {NuxtRuntimeConfig} from "@nuxt/types/config/runtime";
import {Route} from "vue-router";
import {Store} from "vuex";
import EnvironmentStatesView from '~/components/EnvironmentStatesView.vue'
import { Component } from '~/types/kronicle-service'

interface Data {
  components: Component[]
}

async function loadData(
  { $config, route, store }: { $config: NuxtRuntimeConfig, route: Route, store: Store<any> }
): Promise<Data> {
  const components = await fetch(
    `${$config.serviceBaseUrl}/v1/components?fields=components(id,name,typeId,tags,description,notes,responsibilities,teams,platformId,states)`
  )
    .then((res) => res.json())
    .then((json) => json.components as Component[])

  store.commit('componentFilters/initialize', {
    components,
    route,
  })

  return {
    components: [] as Component[],
  }
}

export default Vue.extend({
  components: {
    EnvironmentStatesView,
  },
  async asyncData({ $config, route, store }) {
    return await loadData({
      $config,
      route,
      store,
    })
  },
  data() {
    return {
      components: [] as Component[],
    }
  },
  head(): MetaInfo {
    return {
      title: 'Kronicle - All Environments',
    }
  },
  mounted() {
    setInterval(
      async () => {
        const data = await loadData({
          $config: this.$config,
          route: this.$route,
          store: this.$store
        })
        this.components = data.components
      },
      60 * 1000
    )
  },
})
</script>
