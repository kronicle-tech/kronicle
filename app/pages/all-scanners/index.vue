<template>
  <div>
    <b-container fluid>
      <b-row>
        <b-col>
          <AllScannersTabs />

          <table
            class="table table-dark table-bordered table-striped mt-2"
            style="width: 100%"
          >
            <thead>
            <tr>
              <th class="scanner-id">Id</th>
              <th class="scanner-description">Description</th>
              <th class="scanner-notes">Notes</th>
            </tr>
            </thead>
            <tbody>
            <tr v-for="scanner in scanners" :key="scanner.id">
              <td class="scanner-id">
                {{ scanner.id }}
              </td>
              <td class="scanner-description">
                <Markdown :markdown="scanner.description" />
              </td>
              <td class="scanner-notes">
                <Markdown :markdown="scanner.notes" />
              </td>
            </tr>
            </tbody>
          </table>
        </b-col>
      </b-row>
    </b-container>
  </div>
</template>

<script lang="ts">
import Vue from 'vue'
import { MetaInfo } from 'vue-meta'
import {BCol, BContainer, BRow} from "bootstrap-vue";
import { Scanner } from '~/types/kronicle-service'
import AllScannersTabs from '~/components/AllScannersTabs.vue'
import Markdown from '~/components/Markdown.vue'

export default Vue.extend({
  components: {
    AllScannersTabs,
    'b-col': BCol,
    'b-container': BContainer,
    'b-row': BRow,
    Markdown,
  },
  async asyncData({ $config }) {
    const scanners = await fetch(
      `${$config.serviceBaseUrl}/v1/scanners?fields=scanners(id,description,notes)`
    )
      .then((res) => res.json())
      .then((json) => json.scanners as Scanner[])

    return {
      scanners,
    }
  },
  data() {
    return {
      scanners: [] as Scanner[],
    }
  },
  head(): MetaInfo {
    return {
      title: 'Kronicle - All Scanners',
    }
  },
})
</script>
