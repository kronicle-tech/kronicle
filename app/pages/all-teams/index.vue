<template>
  <div>
    <b-container fluid>
      <b-row>
        <b-col>
          <AllTeamsTabs />

          <b-alert show="10" dismissible variant="info" class="my-3">
            Click a team's name in the table below to view more information about that
            team
          </b-alert>

          <table
            class="table table-dark table-bordered table-striped mt-2"
            style="width: 100%"
          >
            <thead>
              <tr>
                <th class="team-name">Name</th>
                <th>Description</th>
              </tr>
            </thead>
            <tbody>
              <tr v-for="team in teams" :key="team.id">
                <td class="team-name table-primary">
                  <TeamName :team="team" />
                </td>
                <td>
                  <Markdown :markdown="team.description" />
                </td>
              </tr>
            </tbody>
          </table>
        </b-col>
      </b-row>
    </b-container>
  </div>
</template>

<script lang="ts">
import Vue from 'vue'
import { MetaInfo } from 'vue-meta'
import {BAlert, BCol, BContainer, BRow} from 'bootstrap-vue'
import { Team } from '~/types/kronicle-service'
import AllTeamsTabs from '~/components/AllTeamsTabs.vue'
import TeamName from '~/components/TeamName.vue'
import Markdown from '~/components/Markdown.vue'

export default Vue.extend({
  components: {
    AllTeamsTabs,
    'b-alert': BAlert,
    'b-col': BCol,
    'b-container': BContainer,
    'b-row': BRow,
    TeamName,
    Markdown,
  },
  async asyncData({ $config }) {
    const teams = await fetch(
      `${$config.serviceBaseUrl}/v1/teams?fields=teams(id,name,emailAddress,description)`
    )
      .then((res) => res.json())
      .then((json) => json.teams as Team[])

    return {
      teams,
    }
  },
  data() {
    return {
      teams: [] as Team[],
    }
  },
  head(): MetaInfo {
    return {
      title: 'Kronicle - All Teams',
    }
  },
})
</script>
