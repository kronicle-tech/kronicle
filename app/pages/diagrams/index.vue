<template>
  <div class="m-3">
    <DiagramsView :diagrams="diagrams" />
  </div>
</template>

<script lang="ts">
import Vue from 'vue'
import { Diagram } from '~/types/kronicle-service'
import DiagramsView from '~/components/DiagramsView.vue'

export default Vue.extend({
  components: {
    DiagramsView,
  },
  async asyncData({ $config }) {
    const diagrams = await fetch(
      `${$config.serviceBaseUrl}/v1/diagrams?fields=diagrams(id,name,type,tags)`
    )
      .then((res) => res.json())
      .then((json) => json.diagrams)

    return {
      diagrams,
    }
  },
  data() {
    return {
      diagrams: [] as Diagram[],
    }
  },
  head() {
    return {
      title: 'Kronicle - Diagrams',
    }
  },
})
</script>
