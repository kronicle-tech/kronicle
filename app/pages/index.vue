<template>
  <div class="m-3">
    <Intro id="intro" :title="introTitle" :markdown="introMarkdown" />

    <b-card-group columns>
      <b-card
        header="Kronicle"
        header-bg-variant="info"
        header-class="lead"
        no-body
      >
        <b-list-group flush>
          <b-list-group-item
            class="d-flex justify-content-between align-items-center text-muted"
          >
            Kronicle contains the following:
          </b-list-group-item>

          <b-list-group-item
            :to="{ name: 'all-areas' }"
            class="d-flex justify-content-between align-items-center"
          >
            <b>Areas</b>
            <span class="lead">
              <b-badge class="area-count" variant="primary" pill>{{
                  areaCount
                }}</b-badge>
            </span>
          </b-list-group-item>

          <b-list-group-item
            :to="{ name: 'all-teams' }"
            class="d-flex justify-content-between align-items-center"
          >
            <b>Teams</b>
            <span class="lead">
              <b-badge class="team-count" variant="primary" pill>{{
                  teamCount
                }}</b-badge>
            </span>
          </b-list-group-item>

          <b-list-group-item
            :to="{ name: 'all-components' }"
            class="d-flex justify-content-between align-items-center"
          >
            <b>Components</b>
            <span class="lead">
              <b-badge class="component-count" variant="primary" pill>{{
                  componentCount
                }}</b-badge>
            </span>
          </b-list-group-item>

          <b-list-group-item
            :to="{ name: 'all-components-tech-debts' }"
            class="d-flex justify-content-between align-items-center"
          >
            <b>Tech Debts</b>
            <span class="lead">
              <b-badge class="tech-debt-count" variant="primary" pill>{{
                  techDebtCount
                }}</b-badge>
            </span>
          </b-list-group-item>

          <b-list-group-item
            :to="{ name: 'all-components-tests' }"
            class="d-flex justify-content-between align-items-center"
          >
            <b>Test Results</b>
            <span class="lead">
              <b-badge class="test-result-count" variant="primary" pill>{{
                  testResultCount
                }}</b-badge>
            </span>
          </b-list-group-item>
        </b-list-group>
      </b-card>

      <b-card
        header="Scanners"
        header-bg-variant="info"
        header-class="lead"
        no-body
      >
        <b-list-group flush>
          <b-list-group-item
            class="d-flex justify-content-between align-items-center text-muted"
          >
            Scanners dynamically add data to Kronicle at runtime:
          </b-list-group-item>

          <b-list-group-item v-for="scanner in scanners" :key="scanner.id">
            <b class="scanner-id text-info">{{ scanner.id }}</b>
            <br />
            <Markdown :markdown="scanner.description" class="description" />
          </b-list-group-item>
        </b-list-group>
      </b-card>

      <b-card
        header="Tests"
        header-bg-variant="info"
        header-class="lead"
        no-body
      >
        <b-list-group flush>
          <b-list-group-item
            class="d-flex justify-content-between align-items-center text-muted"
          >
            Tests evaluate the data associated with components in Kronicle and
            return outcomes of pass, fail and not applicable
          </b-list-group-item>

          <b-list-group-item
            v-for="test in tests"
            :key="test.id"
            :to="{
              name: 'tests-testId',
              params: { testId: test.id },
            }"
          >
            <b class="test-id text-info">{{ test.id }}</b>
            <br />
            <Markdown :markdown="test.description" class="description" />
          </b-list-group-item>
        </b-list-group>
      </b-card>

      <b-card
        header="Platforms"
        header-bg-variant="info"
        header-class="lead"
        no-body
      >
        <b-list-group-item
          class="d-flex justify-content-between align-items-center text-muted"
        >
          The following platforms are associated with components in Kronicle:
        </b-list-group-item>

        <b-list-group flush>
          <b-list-group-item
            v-for="platformCount in platformCounts"
            :key="platformCount.item"
            :to="{
              name: 'all-components',
              query: { platformId: platformCount.item || 'undefined' },
            }"
            class="d-flex justify-content-between align-items-center"
          >
            <b class="platform">{{ platformCount.item || 'missing' }}</b>
            <span class="lead">
              <b-badge variant="primary" pill>
                {{ platformCount.count }}
              </b-badge>
            </span>
          </b-list-group-item>
        </b-list-group>
      </b-card>

      <b-card
        header="Tags"
        header-bg-variant="info"
        header-class="lead"
        no-body
      >
        <b-list-group flush>
          <b-list-group-item
            class="d-flex justify-content-between align-items-center text-muted"
          >
            The components in Kronicle are tagged with the following:
          </b-list-group-item>

          <b-list-group-item
            v-for="tagCount in tagCounts"
            :key="tagCount.item"
            :to="{
              name: 'all-components',
              query: { tag: tagCount.item },
            }"
            class="d-flex justify-content-between align-items-center"
          >
            <b class="tag">{{ tagCount.item }}</b>
            <span class="lead">
              <b-badge variant="primary" pill>
                {{ tagCount.count }}
              </b-badge>
            </span>
          </b-list-group-item>
        </b-list-group>
      </b-card>

      <b-card
        header="Component Types"
        header-bg-variant="info"
        header-class="lead"
        no-body
      >
        <b-list-group flush>
          <b-list-group-item
            class="d-flex justify-content-between align-items-center text-muted"
          >
            There are the following types of components in Kronicle:
          </b-list-group-item>

          <b-list-group-item
            v-for="componentTypeCount in componentTypeCounts"
            :key="componentTypeCount.item"
            :to="{
              name: 'all-components',
              query: { componentType: componentTypeCount.item },
            }"
            class="d-flex justify-content-between align-items-center"
          >
            <b class="component-type">{{ componentTypeCount.item }}</b>
            <span class="lead">
              <b-badge variant="primary" pill>
                {{ componentTypeCount.count }}
              </b-badge>
            </span>
          </b-list-group-item>
        </b-list-group>
      </b-card>
    </b-card-group>
  </div>
</template>

<style scoped>
.description {
  display: inline-block;
  color: white;
  text-decoration: none;
}

.description >>> p {
  margin-bottom: 0;
}
</style>

<script lang="ts">
import Vue from 'vue'
import {
  BBadge,
  BCard,
  BCardGroup,
  BListGroup,
  BListGroupItem,
} from 'bootstrap-vue'
import { MetaInfo } from 'vue-meta'
import { Area, Component, Scanner, Team, Test } from '~/types/kronicle-service'
import { ItemCount, itemCounts } from '~/src/arrayUtils'
import Intro from '~/components/Intro.vue'
import Markdown from '~/components/Markdown.vue'

export default Vue.extend({
  components: {
    'b-badge': BBadge,
    'b-card': BCard,
    'b-card-group': BCardGroup,
    'b-list-group': BListGroup,
    'b-list-group-item': BListGroupItem,
    Intro,
    Markdown,
  },
  async asyncData({ $config }) {
    const areas = await fetch(
      `${$config.serviceBaseUrl}/v1/areas?fields=areas(id)`
    )
      .then((res) => res.json())
      .then((json) => json.areas as Area[])

    const teams = await fetch(
      `${$config.serviceBaseUrl}/v1/teams?fields=teams(id)`
    )
      .then((res) => res.json())
      .then((json) => json.teams as Team[])

    const components = await fetch(
      `${$config.serviceBaseUrl}/v1/components?fields=components(id,typeId,platformId,tags,techDebts(doesNotExist),testResults(doesNotExist))`
    )
      .then((res) => res.json())
      .then((json) => json.components as Component[])

    const scanners = await fetch(
      `${$config.serviceBaseUrl}/v1/scanners?fields=scanners(id,description)`
    )
      .then((res) => res.json())
      .then((json) => json.scanners as Scanner[])

    const tests = await fetch(
      `${$config.serviceBaseUrl}/v1/tests?fields=tests(id,description)`
    )
      .then((res) => res.json())
      .then((json) => json.tests as Test[])

    return {
      areas,
      teams,
      components,
      scanners,
      tests,
    }
  },
  data() {
    return {
      introTitle: this.$config.introTitle as String | undefined,
      introMarkdown: this.$config.introMarkdown as String | undefined,
      areas: [] as Area[],
      teams: [] as Team[],
      components: [] as Component[],
      scanners: [] as Scanner[],
      tests: [] as Test[],
    }
  },
  head(): MetaInfo {
    return {
      title: 'Kronicle',
    }
  },
  computed: {
    areaCount(): number {
      return this.areas.length
    },
    teamCount(): number {
      return this.teams.length
    },
    componentCount(): number {
      return this.components.length
    },
    componentTypeCounts(): ItemCount<string>[] {
      return itemCounts(this.components.map((component) => component.typeId))
    },
    platformCounts(): ItemCount<string>[] {
      return itemCounts(
        this.components.map((component) => component.platformId)
      )
    },
    tagCounts(): ItemCount<string>[] {
      return itemCounts(
        this.components.flatMap((component) => component.tags ?? [])
      )
    },
    techDebtCount(): number {
      return this.components.flatMap((component) => component.techDebts ?? [])
        .length
    },
    testResultCount(): number {
      return this.components.flatMap((component) => component.testResults ?? [])
        .length
    },
  },
})
</script>
