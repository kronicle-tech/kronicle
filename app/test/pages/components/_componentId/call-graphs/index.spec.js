import Index from '~/pages/components/_componentId/call-graphs/index.vue'
import { createPageWrapper } from '~/test/pages/pageUtils'
import {
  createComponent,
  createDependency,
  createSubComponentNode,
} from '~/test/testDataUtils'
import {
  expectCallGraphCount,
  expectNodeCount,
  expectNodeVariants,
} from '~/test/callGraphUtils'

describe('Index', () => {
  const route = {
    params: {
      componentId: 'test-component-id-1',
    },
  }
  let component
  let nodes
  let callGraphs
  let wrapperActions
  let wrapper
  async function createWrapper() {
    wrapper = await createPageWrapper(Index, {
      route,
      serviceRequests: {
        '/v1/components/test-component-id-1?fields=component(id,name)': {
          responseBody: { component },
        },
        '/v1/components/test-component-id-1/nodes': {
          responseBody: { nodes },
        },
        '/v1/components/test-component-id-1/call-graphs': {
          responseBody: { callGraphs },
        },
      },
    })
    for (const wrapperAction of wrapperActions) {
      await wrapperAction(wrapper)
    }
  }

  beforeEach(() => {
    component = createComponent({ componentNumber: 1 })
    nodes = []
    callGraphs = []
    wrapperActions = []
  })

  afterEach(() => {
    wrapper.destroy()
    wrapper = null
  })

  test('has the right page title', async () => {
    await createWrapper()
    expect(wrapper.vm.$metaInfo.title).toBe(
      'Kronicle - Test Component Name 1 - Call Graphs'
    )
  })

  describe('when Get Component Nodes service endpoint returns no nodes', () => {
    test('renders a message saying there are no call graphs available', async () => {
      await createWrapper()
      expect(wrapper.element).toMatchSnapshot()
      expect(wrapper.text()).toEqual(
        expect.stringContaining(
          'No call graphs are available for this component'
        )
      )
    })
  })

  describe('when Get Component Nodes service endpoints returns nodes and Get Component Call Graphs service endpoint returns call graphs', () => {
    beforeEach(() => {
      nodes = [
        createSubComponentNode({
          componentNodeNumber: 1,
          subComponentNodeNumber: 1,
        }),
        createSubComponentNode({
          componentNodeNumber: 1,
          subComponentNodeNumber: 2,
        }),
      ]
      callGraphs = [
        {
          nodes: [
            createSubComponentNode({
              componentNodeNumber: 1,
              subComponentNodeNumber: 1,
            }),
            createSubComponentNode({ componentNodeNumber: 2 }),
          ],
          dependencies: [createDependency({ sourceIndex: 0, targetIndex: 1 })],
          traceCount: 1,
        },
        {
          nodes: [
            createSubComponentNode({
              componentNodeNumber: 1,
              subComponentNodeNumber: 2,
            }),
            createSubComponentNode({ componentNodeNumber: 2 }),
          ],
          dependencies: [createDependency({ sourceIndex: 0, targetIndex: 1 })],
          traceCount: 1,
        },
        {
          nodes: [
            createSubComponentNode({
              componentNodeNumber: 1,
              subComponentNodeNumber: 2,
            }),
            createSubComponentNode({ componentNodeNumber: 3 }),
          ],
          dependencies: [createDependency({ sourceIndex: 0, targetIndex: 1 })],
          traceCount: 1,
        },
      ]
    })

    test('renders the nodes and call graphs', async () => {
      await createWrapper()
      expect(wrapper.element).toMatchSnapshot()
      expectCallGraphCount(wrapper, 1)
      expectNodeCount(wrapper, 2)
      expectNodeVariants(wrapper, ['success', 'secondary'])
    })
  })
})
