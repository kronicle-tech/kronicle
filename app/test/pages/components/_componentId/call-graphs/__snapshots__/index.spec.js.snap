// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index when Get Component Diagrams service endpoint returns no diagrams renders a message saying there are no call graphs available 1`] = `
<div
  class="m-3"
>
  <h1
    class="text-info my-3"
  >
    Test Component Name 1 - Call Graphs
  </h1>
   
  <div
    class="my-3"
  >
    <a
      class="btn my-1 btn-secondary"
      href="/components/test-component-id-1"
      target="_self"
    >
       Info 
    </a>
     
    <a
      class="btn my-1 btn-secondary"
      href="/components/test-component-id-1/diagrams"
      target="_self"
    >
      
    Diagrams
  
    </a>
     
    <!---->
     
    <!---->
     
    <!---->
     
    <!---->
     
    <a
      class="btn my-1 btn-secondary"
      href="/components/test-component-id-1/cross-functional-requirements"
      target="_self"
    >
      
    Cross Functional Requirements
  
    </a>
     
    <a
      class="btn my-1 btn-secondary"
      href="/components/test-component-id-1/tech-debts"
      target="_self"
    >
      
    Tech Debts
  
    </a>
     
    <a
      class="btn my-1 btn-secondary"
      href="/components/test-component-id-1/errors"
      target="_self"
    >
      
    Errors
  
    </a>
     
    <!---->
     
    <!---->
     
    <!---->
     
    <!---->
     
    <!---->
     
    <a
      class="btn my-1 btn-secondary"
      href="/components/test-component-id-1/lines-of-code"
      target="_self"
    >
      
    Lines of Code
  
    </a>
     
    <!---->
     
    <!---->
     
    <!---->
     
    <a
      class="btn my-1 btn-secondary"
      href="/components/test-component-id-1/tests"
      target="_self"
    >
      
    Tests
  
    </a>
  </div>
   
  <div>
    <transition-stub
      class="my-3"
      css="true"
      enteractiveclass=""
      enterclass=""
      entertoclass="show"
      leaveactiveclass=""
      leaveclass="show"
      leavetoclass=""
      name=""
    >
      <div
        aria-atomic="true"
        aria-live="polite"
        class="alert alert-dismissible alert-info"
        role="alert"
      >
        <button
          aria-label="Close"
          class="close"
          type="button"
        >
          ×
        </button>
        <p>
          
      The call graphs on this page come from a combination of Zipkin traces.
      The visualisation is not out-of-the-box Zipkin functionality and is
      bespoke to Kronicle.
    
        </p>
         
        <p
          class="mb-0"
        >
          
      A maximum of 20 call graphs are displayed on this
      page at any one time.
    
        </p>
      </div>
    </transition-stub>
     
    <div>
      <transition-stub
        class="my-3"
        css="true"
        enteractiveclass=""
        enterclass=""
        entertoclass="show"
        leaveactiveclass=""
        leaveclass="show"
        leavetoclass=""
        name=""
      >
        <div
          aria-atomic="true"
          aria-live="polite"
          class="alert alert-warning"
          role="alert"
        >
          <!---->
          
      No call graphs are available for this component. Possibly:

      
          <ol>
            <li>
              The component is not integrated with Zipkin yet or
            </li>
             
            <li>
              
          Kronicle may have restarted recently and not have loaded Zipkin data
          yet
        
            </li>
          </ol>
        </div>
      </transition-stub>
    </div>
     
    <div
      class="call-graphs"
    />
     
    <div
      class="nodes"
    >
      <h4>
        Component Nodes
      </h4>
       
    </div>
  </div>
</div>
`;

exports[`Index when Get Diagrams service endpoints returns a call graph renders the nodes and call graphs 1`] = `
<div
  class="m-3"
>
  <h1
    class="text-info my-3"
  >
    Test Component Name 1 - Call Graphs
  </h1>
   
  <div
    class="my-3"
  >
    <a
      class="btn my-1 btn-secondary"
      href="/components/test-component-id-1"
      target="_self"
    >
       Info 
    </a>
     
    <a
      class="btn my-1 btn-secondary"
      href="/components/test-component-id-1/diagrams"
      target="_self"
    >
      
    Diagrams
  
    </a>
     
    <!---->
     
    <!---->
     
    <!---->
     
    <!---->
     
    <a
      class="btn my-1 btn-secondary"
      href="/components/test-component-id-1/cross-functional-requirements"
      target="_self"
    >
      
    Cross Functional Requirements
  
    </a>
     
    <a
      class="btn my-1 btn-secondary"
      href="/components/test-component-id-1/tech-debts"
      target="_self"
    >
      
    Tech Debts
  
    </a>
     
    <a
      class="btn my-1 btn-secondary"
      href="/components/test-component-id-1/errors"
      target="_self"
    >
      
    Errors
  
    </a>
     
    <!---->
     
    <!---->
     
    <!---->
     
    <!---->
     
    <!---->
     
    <a
      class="btn my-1 btn-secondary"
      href="/components/test-component-id-1/lines-of-code"
      target="_self"
    >
      
    Lines of Code
  
    </a>
     
    <!---->
     
    <!---->
     
    <!---->
     
    <a
      class="btn my-1 btn-secondary"
      href="/components/test-component-id-1/tests"
      target="_self"
    >
      
    Tests
  
    </a>
  </div>
   
  <div>
    <transition-stub
      class="my-3"
      css="true"
      enteractiveclass=""
      enterclass=""
      entertoclass="show"
      leaveactiveclass=""
      leaveclass="show"
      leavetoclass=""
      name=""
    >
      <div
        aria-atomic="true"
        aria-live="polite"
        class="alert alert-dismissible alert-info"
        role="alert"
      >
        <button
          aria-label="Close"
          class="close"
          type="button"
        >
          ×
        </button>
        <p>
          
      The call graphs on this page come from a combination of Zipkin traces.
      The visualisation is not out-of-the-box Zipkin functionality and is
      bespoke to Kronicle.
    
        </p>
         
        <p
          class="mb-0"
        >
          
      A maximum of 20 call graphs are displayed on this
      page at any one time.
    
        </p>
      </div>
    </transition-stub>
     
    <!---->
     
    <div
      class="call-graphs"
    >
      <div
        class="call-graph my-2"
      >
        <h5>
          
        Call Graph 1
        
          <span
            class="badge ml-2 badge-primary"
          >
            
          1 Traces
        
          </span>
        </h5>
         
        <svg
          height="1200"
          overflow="auto"
          width="100%"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            class="component-dependency-graph-background"
            fill-opacity="0"
          />
           
          <defs>
            <marker
              class="edge-marker all-edge-marker"
              id="component-dependency-graph-all-edge-marker"
              markerHeight="11.5"
              markerUnits="userSpaceOnUse"
              markerWidth="7.1"
              orient="auto-start-reverse"
              refX="18"
              refY="5.75"
              viewBox="0 0 7.1 11.5"
            >
              <path
                d="M1 11.5L0 10.4L5.1 5.7L0 1L1 0L7.1 5.7L1 11.5"
              />
            </marker>
            <marker
              class="edge-marker other-edge-marker"
              id="component-dependency-graph-other-edge-marker"
              markerHeight="11.5"
              markerUnits="userSpaceOnUse"
              markerWidth="7.1"
              orient="auto-start-reverse"
              refX="18"
              refY="5.75"
              viewBox="0 0 7.1 11.5"
            >
              <path
                d="M1 11.5L0 10.4L5.1 5.7L0 1L1 0L7.1 5.7L1 11.5"
              />
            </marker>
            <marker
              class="edge-marker scope-related-edge-marker"
              id="component-dependency-graph-scope-related-edge-marker"
              markerHeight="11.5"
              markerUnits="userSpaceOnUse"
              markerWidth="7.1"
              orient="auto-start-reverse"
              refX="18"
              refY="5.75"
              viewBox="0 0 7.1 11.5"
            >
              <path
                d="M1 11.5L0 10.4L5.1 5.7L0 1L1 0L7.1 5.7L1 11.5"
              />
            </marker>
            <marker
              class="edge-marker scoped-edge-marker"
              id="component-dependency-graph-scoped-edge-marker"
              markerHeight="11.5"
              markerUnits="userSpaceOnUse"
              markerWidth="7.1"
              orient="auto-start-reverse"
              refX="18"
              refY="5.75"
              viewBox="0 0 7.1 11.5"
            >
              <path
                d="M1 11.5L0 10.4L5.1 5.7L0 1L1 0L7.1 5.7L1 11.5"
              />
            </marker>
            <marker
              class="edge-marker related-edge-marker"
              id="component-dependency-graph-related-edge-marker"
              markerHeight="11.5"
              markerUnits="userSpaceOnUse"
              markerWidth="7.1"
              orient="auto-start-reverse"
              refX="18"
              refY="5.75"
              viewBox="0 0 7.1 11.5"
            >
              <path
                d="M1 11.5L0 10.4L5.1 5.7L0 1L1 0L7.1 5.7L1 11.5"
              />
            </marker>
            <marker
              class="edge-marker direct-edge-marker"
              id="component-dependency-graph-direct-edge-marker"
              markerHeight="11.5"
              markerUnits="userSpaceOnUse"
              markerWidth="7.1"
              orient="auto-start-reverse"
              refX="18"
              refY="5.75"
              viewBox="0 0 7.1 11.5"
            >
              <path
                d="M1 11.5L0 10.4L5.1 5.7L0 1L1 0L7.1 5.7L1 11.5"
              />
            </marker>
            <marker
              class="edge-marker selected-edge-marker"
              id="component-dependency-graph-selected-edge-marker"
              markerHeight="11.5"
              markerUnits="userSpaceOnUse"
              markerWidth="7.1"
              orient="auto-start-reverse"
              refX="18"
              refY="5.75"
              viewBox="0 0 7.1 11.5"
            >
              <path
                d="M1 11.5L0 10.4L5.1 5.7L0 1L1 0L7.1 5.7L1 11.5"
              />
            </marker>
          </defs>
           
          <g />
          <g />
          <g />
          <g />
          <g />
          <g>
            <path
              class="edge direct-edge"
              d="M50,50L50,250"
              id="component-dependency-graph-edge-0"
              marker-end="url(#component-dependency-graph-direct-edge-marker)"
            />
          </g>
          <g />
           
          <g />
          <g />
          <g />
          <g />
          <g />
          <g />
          <g>
            <circle
              class="node selected-node"
              cx="50"
              cy="50"
              id="component-dependency-graph-node-0"
              r="10"
              title="test-component-id-1, test-span-name-1-1"
            />
            <circle
              class="node selected-node"
              cx="50"
              cy="250"
              id="component-dependency-graph-node-1"
              r="10"
              title="test-component-id-1, test-span-name-2-2"
            />
          </g>
           
          <g />
          <g />
          <g />
          <g />
          <g />
          <g />
          <g>
            <text
              class="node-label selected-node-label"
              font-size="14"
              stroke-width="1.75"
              x="67"
              y="70"
            >
              <tspan
                x="67"
                y="70"
              >
                
        test-component-id-1
      
              </tspan>
              <tspan
                x="67"
                y="86.8"
              >
                
        test-span-name-1-1
      
              </tspan>
            </text>
            <text
              class="node-label selected-node-label"
              font-size="14"
              stroke-width="1.75"
              x="67"
              y="270"
            >
              <tspan
                x="67"
                y="270"
              >
                
        test-component-id-1
      
              </tspan>
              <tspan
                x="67"
                y="286.8"
              >
                
        test-span-name-2-2
      
              </tspan>
            </text>
          </g>
        </svg>
      </div>
    </div>
     
    <div
      class="nodes"
    >
      <h4>
        Component Nodes
      </h4>
       
      <button
        class="btn node my-1 btn-success btn-block"
        type="button"
      >
        
      test-span-name-1-1
      
        <!---->
      </button>
      <button
        class="btn node my-1 btn-secondary btn-block"
        type="button"
      >
        
      test-span-name-2-2
      
        <!---->
      </button>
    </div>
  </div>
</div>
`;
